apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo/cache/normalized/internal/CacheMapBuilder.kt

```diff
@@ -0,0 +1,116 @@
+package com.apollographql.apollo.cache.normalized.internal
+
+import com.apollographql.apollo.api.Operation
+import com.apollographql.apollo.api.ResponseField
+import com.apollographql.apollo.cache.CacheHeaders
+import com.apollographql.apollo.cache.normalized.CacheKey
+import com.apollographql.apollo.cache.normalized.CacheKeyResolver
+import com.apollographql.apollo.cache.normalized.CacheReference
+
+class CacheMapBuilder(
+    private val readableStore: ReadableStore,
+    private val rootKey: String,
+    private val variables: Operation.Variables,
+    private val cacheKeyResolver: CacheKeyResolver,
+    private val cacheHeaders: CacheHeaders,
+    private val rootResponseFields: Map<String, Array<ResponseField>>
+) {
+  private val cacheKeyBuilder = RealCacheKeyBuilder()
+
+  class PendingReference(
+      val key: String,
+      val possibleResponseFields: Map<String, Array<ResponseField>>
+  )
+
+  private val fieldSets = mutableMapOf<String, Map<String, Any?>>()
+
+  private val pendingReferences = mutableListOf<PendingReference>()
```

Comment:
nit: I guess it can be local variable in `toMap()` function

Reply:
I guess the same applicable for fieldSets

Reply:
> I guess it can be local variable in toMap() function

And passing it as parameter down the call chain ? That might be a bit slower but allow to reuse the `CacheMapBuilder`. On the other hand, we're only using it once every time 🤷

Reply:
Yeah not a big deal TBH, just to avoid global (to class) mutable state. But.... it's really a nit, up to you.